src/main/scala/edu/berkeley/ce/sparkrocks/SeedJointSelector.scala

```diff
@@ -3,242 +3,221 @@ package edu.berkeley.ce.sparkrocks
+    *         be further subdivided. Otherwise, returns None.
     */
-  private def testVolumes(joint1: Joint, joint2: Joint, inputBlock: Block,
-                          desiredVolume: Double, totalVolume: Double,
-                          tolerance: Double): Option[(Joint, Block)] = {
-    val blockSet1 = inputBlock cut joint1
-    val blockSet2 = inputBlock cut joint2
-
-    val nonRedundantBlocks1 = blockSet1 map { case block @ Block(blockCenter, _, _) =>
-        Block(blockCenter, block.nonRedundantFaces)
-    }
-    val nonRedundantBlocks2 = blockSet2 map { case block @ Block(blockCenter, _, _) =>
+  def testVolumes(joint: Joint, initialVolume: Block, desiredVolume: Double): Option[Block] = {
+    val blocks = initialVolume cut joint
+    val nonRedundantBlocks = blocks map { case block @ Block(blockCenter, _, _) =>
       Block(blockCenter, block.nonRedundantFaces)
     }
 
     // Sort generated blocks based on volume
-    val sortedBlocks1 = nonRedundantBlocks1.sortWith(_.volume < _.volume)
-    val sortedBlocks2 = nonRedundantBlocks2.sortWith(_.volume < _.volume)
+    val sortedBlocks = nonRedundantBlocks.sortWith(_.volume < _.volume)
 
-    if (sortedBlocks1.length == 1) {
-      // joint1 does not intersect the block
-      if (sortedBlocks2.length == 1) {
-        // Neither joint intersects the block
-        None
-      } else if (sortedBlocks2(0).volume >= desiredVolume) {
-        // joint2 sufficiently divides block, remaining volume should still be further subdivided
-        Some(joint2, sortedBlocks2(1))
-      } else if (sortedBlocks2(1).volume <= desiredVolume) {
-        // joint2 sufficiently divides block, remaining volume requires no further subdivision
-        Some(joint2, sortedBlocks2(1))
-      } else {
-        // joint 2 does not divide block adequately
-        None
-      }
-    } else if (sortedBlocks2.length == 1) {
-      // joint 2 does not intersect the block
-      if (sortedBlocks1(0).volume >= desiredVolume) {
-        // joint1 sufficiently divides block, remaining volume should still be further subdivided
-        Some(joint1, sortedBlocks1(1))
-      } else if (sortedBlocks1(1).volume <= desiredVolume) {
-        // joint1 sufficiently divides block, remaining volume requires no further subdivision
-        Some(joint1, sortedBlocks1(1))
-      } else {
-        // joint 1 does not divide the block adequately
-        None
-      }
+    // Joint does not intersect the block
+    if (sortedBlocks.length == 1) {
+      None
     } else {
-      // BOTH joints intersect the block
-      // Compare volumes to desired volumes
-      val volumeDiff1 = (sortedBlocks1(0).volume - desiredVolume) / desiredVolume
-      val volumeDiff2 = (sortedBlocks2(0).volume - desiredVolume) / desiredVolume
+      val smallBlockDiff = math.abs(sortedBlocks(0).volume - desiredVolume) / desiredVolume
+      val largeBlockDiff = math.abs(sortedBlocks(1).volume - desiredVolume) / desiredVolume
```

Comment:
Is `largeBlockDiff` actually used anywhere? I couldn't find any cases, but I could be missing something obvious.

Have you run these changes through IntelliJ to see if there are any warnings to clean up?